homeassistant/components/qnap/sensor.py

```diff
@@ -1,346 +1,323 @@
         native_unit_of_measurement=UnitOfInformation.GIBIBYTES,
-        device_class=SensorDeviceClass.DATA_SIZE,
         icon="mdi:memory",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
     ),
-    SensorEntityDescription(
+    QNapSensorEntityDescription(
+        stype="memory",
         key="memory_used",
         name="Memory Used",
         native_unit_of_measurement=UnitOfInformation.GIBIBYTES,
-        device_class=SensorDeviceClass.DATA_SIZE,
         icon="mdi:memory",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
     ),
-    SensorEntityDescription(
+    QNapSensorEntityDescription(
+        stype="memory",
         key="memory_percent_used",
         name="Memory Usage",
         native_unit_of_measurement=PERCENTAGE,
         icon="mdi:memory",
+        entity_registry_enabled_default=True,
+        state_class=SensorStateClass.MEASUREMENT,
     ),
-)
-_NETWORK_MON_COND: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
+    QNapSensorEntityDescription(
+        stype="network",
         key="network_link_status",
         name="Network Link",
         icon="mdi:checkbox-marked-circle-outline",
+        entity_registry_enabled_default=True,
     ),
-    SensorEntityDescription(
+    QNapSensorEntityDescription(
+        stype="network",
         key="network_tx",
         name="Network Up",
         native_unit_of_measurement=UnitOfDataRate.MEBIBYTES_PER_SECOND,
-        device_class=SensorDeviceClass.DATA_RATE,
         icon="mdi:upload",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
     ),
-    SensorEntityDescription(
+    QNapSensorEntityDescription(
+        stype="network",
         key="network_rx",
         name="Network Down",
         native_unit_of_measurement=UnitOfDataRate.MEBIBYTES_PER_SECOND,
-        device_class=SensorDeviceClass.DATA_RATE,
         icon="mdi:download",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
     ),
-)
-_DRIVE_MON_COND: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
+    QNapSensorEntityDescription(
+        stype="drive",
         key="drive_smart_status",
         name="SMART Status",
         icon="mdi:checkbox-marked-circle-outline",
+        entity_registry_enabled_default=False,
     ),
-    SensorEntityDescription(
+    QNapSensorEntityDescription(
+        stype="drive",
         key="drive_temp",
         name="Temperature",
         native_unit_of_measurement=UnitOfTemperature.CELSIUS,
-        device_class=SensorDeviceClass.TEMPERATURE,
+        icon="mdi:thermometer",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
     ),
-)
-_VOLUME_MON_COND: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
+    QNapSensorEntityDescription(
```

Comment:
These are new sensors. I suggest that you create a specific PR to add these two new sensors.

Reply:
Folder sensors removed.